### ISSUE TITLE
Array.CreateInstance(<type>, System.Int32[]) results in 'TypeError: No method matches given arguments' in v2.5.0

### ISSUE THREAD

**Issue by pkthong** [pythonnet/pythonnet#1187]
### Environment

-   Pythonnet version: 2.5.0
-   Python version: 3.6.9
-   Operating System: Windows 10

### Details

-   Describe what you were trying to get done.

   Array.CreateInstance for multi-dimensional arrays worked fine in v2.4.0. After upgrading to v2.5.0+, an error message gets thrown. 

-   What commands did you run to trigger this issue? 

```python
import numpy as np
import ctypes
import clr, System
from System import Array, Int32
from System.Runtime.InteropServices import GCHandle, GCHandleType
import sys
import os


_MAP_NP_NET = {
    np.dtype('float32'): System.Single,
    np.dtype('float64'): System.Double,
    np.dtype('int8')   : System.SByte,
    np.dtype('int16')  : System.Int16,
    np.dtype('int32')  : System.Int32,
    np.dtype('int64')  : System.Int64,
    np.dtype('uint8')  : System.Byte,
    np.dtype('uint16') : System.UInt16,
    np.dtype('uint32') : System.UInt32,
    np.dtype('uint64') : System.UInt64,
    np.dtype('bool')   : System.Boolean,
}

def asNetArray(npArray):
    dims = npArray.shape
    dtype = npArray.dtype

    netDims = Array.CreateInstance(Int32, npArray.ndim)
    for I in range(npArray.ndim):
        netDims[I] = int(dims[I])

    if not npArray.flags.c_contiguous:
        npArray = npArray.copy(order='C')
    assert npArray.flags.c_contiguous

    try:
        netArray = Array.CreateInstance(_MAP_NP_NET[dtype], netDims)
    except KeyError:
        raise NotImplementedError("asNetArray does not yet support dtype {}".format(dtype))

    try: # Memmove
        destHandle = GCHandle.Alloc(netArray, GCHandleType.Pinned)
        sourcePtr = npArray.__array_interface__['data'][0]
        destPtr = destHandle.AddrOfPinnedObject().ToInt64()
        ctypes.memmove(destPtr, sourcePtr, npArray.nbytes)
    finally:
        if destHandle.IsAllocated: destHandle.Free()
    return netArray

if __name__ == '__main__':
    foo = np.full([1024,1024], 2.5, dtype=np.int32)

    netFoo = asNetArray( foo )

```

-   If there was a crash, please include the traceback here.

```python
Traceback (most recent call last):
  File "pythonnetUtils.py", line 53, in <module>
    netFoo = asNetArray( foo )
  File "pythonnetUtils.py", line 37, in asNetArray
    netArray = Array.CreateInstance(_MAP_NP_NET[dtype], netDims)
TypeError: No method matches given arguments for CreateInstance: (<class 'CLR.CLR Metatype'>, <class 'System.Int32[]'>)
```

**Comment by lostmsu** [comment]
Hm, @pkthong can you double check your code? When I do `str(Int32)` in the latest build, I get `<class 'System.Int32'>`, not `<class 'CLR.CLR Metatype'>`.

**Comment by pkthong** [comment]
Python 3.8.0 (default, Nov  6 2019, 16:00:02) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> print(sys.version)
3.8.0 (default, Nov  6 2019, 16:00:02) [MSC v.1916 64 bit (AMD64)]
>>> import clr, System
>>> print(type(System.Int32))
<class 'CLR.CLR Metatype'>
>>> print(clr.__version__)
3.0.0dev

**Comment by filmor** [comment]
Hmm, I'll give that a try locally. The "type" it lists there is correct for Python, `CLR.CLR Metatype` (awful name...) is the Python `type` for `System.Type` objects, but it seems like the method binding doesn't make that link anymore.

**Comment by pkthong** [comment]
Running through the changes between v2.4.0 and v2.5.0, the last commit that works for the above test is (4a92d80)

**Comment by filmor** [comment]
Okay, then it's indeed https://github.com/pythonnet/pythonnet/pull/1106. That change actually improved things, in that it allows "proper" use of `params` arrays (like `CreateInstance` uses). You can fix your code by changing

        netArray = Array.CreateInstance(_MAP_NP_NET[dtype], netDims)

to

        netArray = Array.CreateInstance(_MAP_NP_NET[dtype], *netDims)

I'm not yet completely sure whether we want to fix it (we definitely want to keep the feature, maybe we can still allow the direct passing of the array as a fallback), but I acknowledge breakage :)

**Comment by pkthong** [comment]
The above suggestion seems to execute, yet accessing the array results in a System.AccessViolationException.

To reproduce this, add the following to the test code.
'''
netFoo2 = Array.Copy(netFoo)
'''

**Comment by robbmcleod** [comment]
I've gone over my code snippet from before and made fixes similar to what @filmor suggested, although it needed an additional change to support complex data types. I put together a gist this time:

https://gist.github.com/robbmcleod/73ca42da5984e6d0e5b6ad28bc4a504e

**Comment by lostmsu** [comment]
I believe this should be fixed in 3.0. Please reopen if not.
